src/sss_client/pam_sss.c

```diff
@@ -1698,7 +1700,13 @@ static int prompt_multi_cert_gdm(pam_handle_t *pamh, struct pam_items *pi)
             ret = ENOMEM;
             goto done;
         }
-        request->list.items[c].key = cai->key_id;
+        ret = asprintf(&cai->choice_list_id, "%zu", c);
```

Comment:
Can it happen `pam_sss()` calls `get_authtok_for_authentication()` -> `prompt_multi_cert()` -> `prompt_multi_cert_gdm()` several times and `cai->choice_list_id` leaks here?

Reply:
Good catch, this might be possible if the first authentication attempt fails, e.g. wrong PIN, and the dialog starts again. Fix in latest version.